pkg/status/workstatus.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/kubestellar/kubestellar/pkg/util"
+)
+
+const workStatusIdentificationIndexKey = "workStatusIdentificationIndex"
+
+type workStatus struct {
+	wecName                string
+	sourceObjectIdentifier util.ObjectIdentifier
+	status                 map[string]interface{}
+}
+
+func (c *Controller) syncWorkStatus(ctx context.Context, key string) error {
+	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+	obj, err := getObject(c.workStatusLister, ns, name)
+	if err != nil {
+		// The resource no longer exist, which means it has been deleted.
+		if errors.IsNotFound(err) {
+			utilruntime.HandleError(fmt.Errorf("object %#v in work queue no longer exists", key))
+			return nil // TODO: this is a bug, should notify the resolver of removal
+		}
+		return err
+	}
+
+	ws, err := convertToWorkStatus(obj)
+	if err != nil {
+		return err
+	}
+	combinedStatusSet := c.combinedStatusResolver.NoteWorkStatus(ws)
+	for combinedStatus := range combinedStatusSet {
+		c.workqueue.AddAfter(combinedStatusRef(combinedStatus.ObjectName.AsNamespacedName().String()), queueingDelay)
+	}
+
+	// only process workstatues with the label for single reported status
+	statusLabelVal, ok := obj.(metav1.Object).GetLabels()[util.BindingPolicyLabelSingletonStatusKey]
+	if !ok {
```

Comment:
Why is this treated differently than `statusLabelVal == util.BindingPolicyLabelSingletonStatusValueUnset` ?

Reply:
This portion of the code is taken from the previous singleton handling logic.

Reply:
Then I am not going to insists on fixing it in this PR.